packages/core/src/modules/sd-jwt-vc/SdJwtVcService.ts

```diff
@@ -409,6 +427,34 @@ export class SdJwtVcService {
 
     const iss = sdJwtVc.jwt.payload['iss'] as string
 
+    if (sdJwtVc.jwt.header?.x5c) {
+      if (!Array.isArray(sdJwtVc.jwt.header.x5c)) {
+        throw new SdJwtVcError('Invalid x5c header in credential. Not an array.')
+      }
+      if (sdJwtVc.jwt.header.x5c.length === 0) {
+        throw new SdJwtVcError('Invalid x5c header in credential. Empty array.')
+      }
+      if (sdJwtVc.jwt.header.x5c.some((x5c) => typeof x5c !== 'string')) {
+        throw new SdJwtVcError('Invalid x5c header in credential. Not an array of strings.')
+      }
+
+      const x509Service = agentContext.dependencyManager.resolve(X509Service)
+      const certificate = x509Service.parseCertificate(sdJwtVc.jwt.header.x5c[0])
+
+      // If the iss value contains a DNS name encoded as a URI using the DNS URI scheme [RFC4501], the DNS name MUST match a dNSName Subject Alternative Name (SAN) [RFC5280] entry of the leaf certificate.
+      if (![...(certificate.sanDnsNames ?? []), ...(certificate.sanUriNames ?? [])].includes(iss)) {
```

Comment:
should sanDnsNames and sanUriNames return an empty array if there's no entries? That would make this code cleaner. Or is there different in meaning from retruning an empty array vs undefined?

Reply:
Technically there is a difference, i.e. undefined = no extension for SAN, [] = extension for SAN was there, but did not contain any values.

I don't think we want to make a distinction between these cases, so I will default to an empty array if nothing is found.